### ISSUE TITLE
Breakpoints do not change

### ISSUE THREAD

**Issue by SoaresMG** [Farfetch/react-context-responsive#11]
## Issue Report

When changing the `breakpoints` nothing happens.

## Expected Behavior

It should be possible to change `breakpoints` and reflect those changes immediately.

## Actual Behavior

Nothing happens, it uses the first `breakpoints` sent.

## Steps to Reproduce the Issue

1. Check [example](https://codesandbox.io/s/fast-dream-codn7)
2. Click the button and check that it still uses the initial breakpoints

**Comment by SoaresMG** [comment]
After investigating it seems that the culprit is `useRef` being used [here](https://github.com/Farfetch/react-context-responsive/blob/master/src/ResponsiveProvider.js#L40), that ref will always have the same value while we are using some memoized functions receiving it as an argument.

**Comment by stephan-noel** [comment]
About this, I was wondering.  Does it make sense to use props for a lot of these things?  Like breakpoints, initialMediaType, a lot actually. I know it's the normal way to parameterize things for React, but usually these things are just configured once for the lifetime of the app.  

How often do you change your breakpoints mid-way through your app?  I think it would help clean up code, improve performance, etc.  I have some ideas on how it can be done.  I'm not sure what's the idiomatic approach in this kind of situation, and it may be eventually be a pretty big overhaul of the API,, but I think it would be an interesting avenue to explore.  What do you think?

**Comment by SoaresMG** [comment]
You are suggesting that those props could be static? Once defined (initially) if they were changed the component wouldn't reflect those changes? 

I see that it may have benefits but we would be sacrificing on how people see component's normal behaviour.
Most people have no need to change breakpoints midway however if someone does it wouldn't be possible to use this component or it would be needed to unmount and mount the component again for example using `key` and IMO it's not something I would like to do.

**Comment by stephan-noel** [comment]
No I'm not suggesting that the props should be static, though one approach would be to rename some props to make this clear.  It is common when dealing with derived state for instance, to consider it "okay" to ignore updates to props if you prefix them with "initial" or something that makes it clear that updates to the prop will be ignored.  

But what I'm suggesting is that these things aren't even part of the normal React data flow.  If I had ResponsiveProvider defined inline in my project, what I would probably do is something like:

```
import breakpoints from ../constants/breakpoints;

const ResponsiveProvider = ({
    initialMediaType,
    defaultOrientation,
    children,
    breakpointsMax,
    mediaQueries,
}) => {

   // use breakpoints here

}
```

So just like we use import variables directly in our component, we could use breakpoints.  We could do something similar but allow it to be configurable by using a closure.  

But yea if someone wanted to change it, as part of React's data flow, it wouldn't be easily achievable, however, I wonder if that would be taking into account a use case that ends up never coming up.  

Maybe I will explore this approach as a POC to elaborate what the advantages are and revisit it in the future.

**Comment by SoaresMG** [comment]
By saying `static` I was referring to being prefixed with `initial`, I understood what you said, the thing is you will lose the ability to change breakpoints (that right now doesn't work anyway) and I can't see how much cleaner the api can get by doing so

**Comment by SoaresMG** [comment]
Also we can't have breakpoints coming as an import since every project has it's own values

**Comment by stephan-noel** [comment]
Yea I'm not sure of how much cleaner it would be, was just exploring ideas.  I wasn't saying to use an import, just showing that there are other ways to get data into a component besides props.  You could just use a function to do the same thing.

But I was just exploring ideas and trying to see which props really should be props, for now I think it's best to leave it as is.

**Comment by filipesmedeiros** [comment]
Hey guys! Can I help with this? I wouldn't mind rewriting `.js`s to `.tsx`s but if you prefer adding `.d.ts`s I could help with that too. I think config and tooling (with Babel and stuff) is what I'd need the most help with, although it should be fine! Let me know if I can start working on this.

**Comment by gsferreira** [comment]
Closed due to project archival #76
